pkg/arvo/app/chat-hook.hoon

```diff
@@ -383,6 +385,52 @@
 |_  bol=bowl:gall
 ++  grp  ~(. grpl bol)
 ::
+++  poke-fix-dms
+  |=  a=%fix-dms
+  ^-  (quip card _state)
+  :_  state
+  %-  zing
+  %+  turn
+    ~(tap by synced)
+  |=  [=path host=ship]
+  ^-  (list card)
+  ?>  ?=([@ @ *] path)
+  ?.  =('~' i.path)
+    ~
+  =/  =ship  (slav %p i.t.path)
+  ?:  =(ship our.bol)
+    ::  local dm, no need to do cleanup
+    ~
+  ?:  ?=(^ (groups-of-chat path))
+    ::  correctly initialized, no need to do cleanup
+    ::
+    ~
+  :-  =-  [%pass /fixdm %agent [our.bol %chat-view] %poke %chat-view-action -]
+      !>  ^-  action:store
+      [%delete path]
```

Comment:
I'm likely missing some context here, so maybe these are dumb questions, but:

Am I reading this correctly in that it will fully delete history for affected DMs?

https://github.com/urbit/urbit/blob/c96705e755f424e90887c656c452ada28797806b/pkg/arvo/app/chat-view.hoon#L235-L240

What's stopping us from keeping them around (in the store, at least), and/or rejoining the foreign DMs?

Reply:
Note that we scry out the messages and move them to the newly created local DM.

Reply:
Oh oops, I missed that. What about reconnecting to the foreign DMs though? Is that off the table because of permission migration issues or similar?